recipes/meson/all/test_package/conanfile.py

```diff
@@ -7,13 +7,15 @@ class TestPackageConan(ConanFile):
     generators = "pkg_config"
 
     def build(self):
-        meson = Meson(self)
-        meson.configure(build_folder="build")
-        meson.build()
+        if not tools.cross_building(self):
+            meson = Meson(self)
+            meson.configure(build_folder="build")
+            meson.build()
```

Comment:
The main meson recipe contains no `settings = "os", "arch"`, and has `def package_id(self): self.info.header_only()`.
So meson should be usable, even when using as a host requirement.

I think the problem this is fixing is the fact that environment variables from host profiles are not inherited.
I would surround the build with `tools.RunEnvironment(self)`. 


```suggestion
        with tools.RunEnvironment(self):
            meson = Meson(self)
            meson.configure(build_folder="build")
            meson.build()
```

Reply:
The underlying problem is that conans.tools.Meson build helper is not yet able to cross compile anything. Will this change somehow put the meson build helper in native compilation mode? If yes, I think this test is not really relevant, because it does not match the settings,  so the package will never be used this way.

Reply:
This applies here too:
https://github.com/conan-io/conan-center-index/pull/7457#discussion_r719392323